test/unit/YieldTokenIncidents/redeemPayout.js

```diff
@@ -573,6 +573,67 @@ describe('redeemPayout', function () {
     }
   });
 
+  it('transfers DAI amount to payoutAddress, as per requested amount and priceBefore', async function () {
+    const { yieldTokenIncidents, assessment, cover, ybEth, dai } = this.contracts;
+    const [member1] = this.accounts.members;
+    const [nonMember1, nonMember2] = this.accounts.nonMembers;
+    const [advisoryBoard] = this.accounts.advisoryBoardMembers;
+    const segment = await getCoverSegment();
+    segment.amount = parseEther('10000');
+
+    await cover.createMockCover(
+      member1.address,
+      2, // productId
+      ASSET.DAI,
+      [segment],
+    );
+
+    {
+      const { timestamp: currentTime } = await ethers.provider.getBlock('latest');
+      await yieldTokenIncidents
+        .connect(advisoryBoard)
+        .submitIncident(2, parseEther('1.1'), currentTime + segment.period / 2, parseEther('100'), '');
+    }
+
+    await assessment.connect(member1).castVote(0, true, parseEther('100'));
+
+    {
+      const { payoutCooldownInDays } = await assessment.config();
+      const { end } = await assessment.getPoll(0);
+      await setTime(end + daysToSeconds(payoutCooldownInDays));
+    }
+
+    await ybEth.connect(member1).approve(yieldTokenIncidents.address, parseEther('10000'));
+
+    // [warning] Cover mock does not subtract the covered amount
+    {
+      const daiBalanceBefore = await dai.balanceOf(member1.address);
+      await yieldTokenIncidents
+        .connect(member1)
+        .redeemPayout(0, 0, 0, parseEther('100'), member1.address, [], { gasPrice: 0 });
+      const daiBalanceAfter = await dai.balanceOf(member1.address);
+      expect(daiBalanceAfter).to.be.equal(daiBalanceBefore.add(parseEther('99')));
```

Comment:
I know other tests here don't do this

But i'd like this 99 to be a computed value based on `priceBefore` and claim amount.

const expectedPayout =. ...